docs/custom-planning-unit-grid-geometry.hld.md

```diff
@@ -0,0 +1,17 @@
+# Goal
+
+Users may sometimes create their own planning unit geometries, instead of
+relying on generated square or hexagonal grids, for a variety of conservation
+reasons, so they should be able to supply their own PU grid via a shapefile
+upload.
+
+# Key assumptions
+
+* shape can be re-uploaded as long as there are no scenarios created yet
+* only `FeatureCollection` with `Polygon` features can be uploaded
```

Comment:
I am very concerned if they end up with irregular shapes that need multipolygons. Also, we need to ensure this feature collection really represents a grid or a surface.